evalml/tests/conftest.py

```diff
@@ -2083,6 +2083,21 @@ def objective_function(self, y_true, y_predicted, X=None):
     return CustomClassificationObjectiveRanges()
 
 
+class CustomBinaryClassificationObjective(BinaryClassificationObjective):
+    _integer_nullable_incompatible = True
+    _boolean_nullable_incompatible = True
+
+    name = "my_objective"
+    greater_is_better = True
+    score_needs_proba = False
+    perfect_score = 1.0
+    is_bounded_like_percentage = True
+    expected_range = [0, 1]
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
+        pass
```

Comment:
oooh, I like that quite a lot! I'll make that change

Reply:
Two things that kept me from implementing this suggestion:
1. The codecove issue can be avoided by replacing `pass` with a string (as the mock above this does - I didntknow why it did that, and now I do lol)
2. testing `_handle_nullable_types` via `objective_function` would only work if we were returning the result of `_handle_nullable_types` in that function. Since we are just using the methods outputs to run the objective function, it wouldn't make sense to test it this way. Once we get to actually calling `_handle_nullable_types` inside of the objective functions, I'll make sure to mock it out and confirm it does get called when we need it, but that wouldn't be part of this PR.